api/src/main/java/io/opentelemetry/OpenTelemetry.java

```diff
@@ -5,264 +5,160 @@
+ *
+ * <p>A global singleton can be retrieved by {@link #get()}. The default for the returned {@link
+ * OpenTelemetry}, if none has been set via {@link #set(OpenTelemetry)} will be created with any
+ * {@link io.opentelemetry.spi.OpenTelemetryFactory}, {@link
+ * io.opentelemetry.trace.spi.TracerProviderFactory}, {@link
+ * io.opentelemetry.metrics.spi.MeterProviderFactory}, or {@link
+ * io.opentelemetry.baggage.spi.BaggageManagerFactory} found on the classpath, or otherwise will be
+ * default, with no-op behavior.
  *
- * <p>The telemetry objects are lazy-loaded singletons resolved via {@link ServiceLoader} mechanism.
+ * <p>If using the OpenTelemetry SDK, you may want to instantiate the {@link OpenTelemetry} to
+ * provide configuration, for example of {@code Resource} or {@code Sampler}. See {@code
+ * OpenTelemetrySdk} and {@code OpenTelemetrySdk.newBuilder} for information on how to construct the
+ * SDK {@link OpenTelemetry}.
  *
  * @see TracerProvider
- * @see MeterProviderFactory
- * @see BaggageManagerFactory
+ * @see MeterProvider
+ * @see BaggageManager
  */
 @ThreadSafe
-public final class OpenTelemetry {
-  private static final Object mutex = new Object();
+public interface OpenTelemetry {
 
-  @Nullable private static volatile OpenTelemetry instance;
+  /**
+   * Returns the registered global {@link OpenTelemetry}. If no call to {@link #set(OpenTelemetry)}
+   * has been made so far, a default {@link OpenTelemetry} composed of functionality any {@link
+   * io.opentelemetry.spi.OpenTelemetryFactory}, {@link
+   * io.opentelemetry.trace.spi.TracerProviderFactory}, {@link
+   * io.opentelemetry.metrics.spi.MeterProviderFactory}, or {@link
+   * io.opentelemetry.baggage.spi.BaggageManagerFactory} found on the classpath, or otherwise * will
+   * be default, with no-op behavior.
+   *
+   * @throws IllegalStateException if a provider has been specified by system property using the
+   *     interface FQCN but the specified provider cannot be found.
+   */
+  static OpenTelemetry get() {
+    return DefaultOpenTelemetry.getGlobalOpenTelemetry();
+  }
 
-  private final TracerProvider tracerProvider;
-  private final MeterProvider meterProvider;
-  private final BaggageManager contextManager;
+  /**
+   * Sets the {@link OpenTelemetry} that should be the global instance. Future calls to {@link
+   * #get()} will return the {@link OpenTelemetry}. This should be called once as early as possible
+   * in your application initialization logic, often in a {@code static} block in your main class.
+   */
+  static void set(OpenTelemetry openTelemetry) {
+    DefaultOpenTelemetry.setGlobalOpenTelemetry(openTelemetry);
+  }
 
-  private volatile ContextPropagators propagators = DefaultContextPropagators.builder().build();
+  /**
+   * Returns the globally registered {@link ContextPropagators} for remote propagation of a context.
+   */
+  static ContextPropagators getGlobalPropagators() {
```

Comment:
I personally prefer the old `getPropagators()` as it's shorter. Should be fine as long as we document it as the globally registered instance, we should be fine.

Reply:
Realized we can't have static and non-static with the same name anyways, ow